CMakePresets.json

```diff
@@ -0,0 +1,528 @@
+{
+  "version": 3,
+  "cmakeMinimumRequired": {
+    "major": 3,
+    "minor": 21,
+    "patch": 1
+  },
+  "configurePresets": [
+    {
+      "name": "linux-config-base",
+      "hidden": true,
+      "generator": "Unix Makefiles",
+      "condition": {
+        "type": "equals",
+        "lhs": "${hostSystemName}",
+        "rhs": "Linux"
+      },
+      "architecture": {
+        "value": "x64",
+        "strategy": "external"
+      },
+      "cacheVariables": {
+        "CMAKE_C_COMPILER": "gcc",
+        "CMAKE_CXX_COMPILER": "g++"
+      }
+    },
+    {
+      "name": "linux-gcc-ubsan-ootree-fast",
+      "displayName": "Linux x64: GCC / Ubsan / Out of tree (fast)",
+      "inherits": "linux-config-base",
+      "binaryDir": "${sourceDir}/.build/${presetName}",
+      "installDir": "${sourceDir}/install",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Ubsan",
+        "CMAKE_C_FLAGS": "-O2",
+        "CMAKE_CXX_FLAGS": "-O2",
+        "RE2C_BUILD_RE2GO": false
+      }
+    },
+    {
+      "name": "linux-gcc-ubsan-ootree-full",
+      "displayName": "Linux x64: GCC / Ubsan / Out of tree (full)",
+      "inherits": "linux-gcc-ubsan-ootree-fast",
+      "installDir": "${sourceDir}/install-2",
+      "cacheVariables": {
+        "RE2C_BUILD_RE2GO": true,
+        "RE2C_BUILD_LIBS": true,
+        "RE2C_REBUILD_LEXERS": true,
+        "RE2C_REBUILD_DOCS": true,
+        "RE2C_FOR_BUILD": "${sourceDir}/install/bin/re2c"
+      }
```

Comment:
Can this block be inherited? As far as I can tell it is duplicated many times (in all "full" presets).

Reply:
Well, more recently, you hinted that there are too many presets :) Of course, we can do more atomic presets and even use multiple inheritance.

Reply:
If I understood you correctly, then here is my patch: https://github.com/skvadrik/re2c/pull/374/commits/2714a357b0b24ea430e53cb2c53f631d48c51aed

Reply:
What you did is exactly what I meant, thanks.